### ISSUE TITLE
Scanning Slow Start When Link Bandwidth Unknown

### ISSUE THREAD

**Issue by zakird** [zmap/zmap#804]
One of the difficulties of using ZMap today is that it's hard to know how fast your underlying network can support scanning. Without ground truth data to compare against, researchers could be missing a sizable fraction of hosts without knowing it, because their send rate is too fast.

One way of addressing this would be to dynamically manage the scan rate, starting slow, and only increasing the scan rate slowly, keeping track that the overall hit rate doesn't decrease as the scan rate increases, up to the set scan rate (which would be considered the max). 

If the hitrate does appear to decrease, we can warn the user, and reduce the scan rate back down to where results were stable.

**Comment by grasstractor** [comment]
Yes, I have encountered this issue in my recent research. Numerous factors significantly impact scanning speed, including time-related variations even among servers with identical hardware and bandwidth configurations from the same cloud service provider. I hope this problem can be effectively addressed.
